app/assets/javascripts/angular/controllers/graph_ctrl.js

```diff
@@ -123,7 +123,8 @@ angular.module("Prometheus.controllers").controller('GraphCtrl',
       var rangeSeconds = Prometheus.Graph.parseDuration(scope.graph.range);
       // bigger denominator == smaller step == more data
       var step = Math.floor(rangeSeconds / scalingFactor)
-      refreshFn(rangeSeconds, step).then(function(data) {
+      var endTime = Math.floor($scope.graph.endTime / 1000);
```

Comment:
Originally, I was going to call `refreshFn()` (on the next line) with updated arguments coming from the user's drag selection. Currently I'm thinking I will just update the `$scope.graph.endTime` and `$scope.graph.range` variables and call for a refresh, though, which makes this unnecessary.

In general I'm attempting to pass variables explicitly instead of defaulting to sending in `$scope` and then picking off what attributes I need, for clarity and to ease the job of extracting the graph drawing into its own independent library (something that is eventually going to happen, so we can consolidate graph functionality between prometheus and promdash).